src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -193,8 +193,8 @@ class DefaultOnOffSceneHandler : public scenes::DefaultSceneHandlerImpl
         {
             auto & decodePair = pair_iterator.GetValue();
             VerifyOrReturnError(decodePair.attributeID == Attributes::OnOff::Id, CHIP_ERROR_INVALID_ARGUMENT);
-            ReturnErrorOnFailure(
-                mSceneEndpointStatePairs.InsertPair(OnOffEndPointPair(endpoint, static_cast<bool>(decodePair.attributeValue))));
+            ReturnErrorOnFailure(mSceneEndpointStatePairs.InsertPair(
+                OnOffEndPointPair(endpoint, static_cast<bool>(decodePair.valueUnsigned8.Value()))));
```

Comment:
What if someone sends malformed input where decodePair.valueUnsigned8 is missing?  Won't this crash?

Reply:
Added a check on `valueUnsigned8.HasValue()` right before.

Note that this is something the PR https://github.com/project-chip/connectedhomeip/pull/32632 will be addressing once the `ValidateAttributePath` method implements the attribute refactor from this PR.